### ISSUE TITLE
Add a changelog.

### ISSUE THREAD

**Issue by LouisGariepy** [cornucopia-rs/cornucopia#190]
Since our main branch is currently sitting at v0.9, and there's a bunch of incoming PRs to merge, now is a good time to add a `CHANGELOG.md` for future releases.

There are a number of different ways to go about this, but notably:
1. Write the changelog manually, following https://keepachangelog.com/en/1.0.0/.
2. Use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to [automatically](https://github.com/google-github-actions/release-please-action) generate a changelog.

<details>

Note that if we go with route 2), I won't be enforcing a pre-commit hook for conventional commits. In my opinion, its the responsibility of the maintainers (not the casual committer) to ensure that proper conventional commits are used. Git provides tools for maintainers to achieve this, notably interactive rebasing.

</details>

---

I'd like to distinguish between "release notes" and "changelog". 

The changelog is meant for developpers or power-users who want to know on a fine-grained level what's happening in the repository and look at the relevant PRs.

Release notes are meant for the average user, explaining at a high-level what's new, what's changed and how to upgrade. By definition they must be written manually, and I'm committed to write these out myself when release time comes, regardless of how we decide to generate the changelog.

**Comment by LouisGariepy** [comment]
I agree that if we go route 2, we should find a way to prevent merging non-conventional commits.

The problem with a required commit hook is that it adds more friction to casual contributors who might not know or care about conventional commits. I don't want to dissuade or slow down PRs just because the author is not familiar with conventional commits.

**Comment by jacobsvante** [comment]
Sounds reasonable 👍🏻

**Comment by Virgiel** [comment]
This is the first time I have heard of conventional commits. As I understand it, each commit has to be formatted in a certain way to be added as a line in the changelog. I'm afraid that this would encourage squashing large changes in a single commit, which might be harder to track.

**Comment by tbillington** [comment]
An alternative to conventional commits is require all PRs also add to the changelog as part of the PR.

**Comment by LouisGariepy** [comment]
@tbillington I'll add this option to the original post! Thanks :)

**Comment by LouisGariepy** [comment]
Completed. Cornucopia now maintains a changelog via release-plz using conventional commits.
